proposals/devcontainer-features.md

```diff
@@ -0,0 +1,143 @@
+
+A feature collection file is a compilation of the `feature.json` files for each individual feature.
+
+If the application finds a `feature-collection.json` file in the root of a downloaded tar file then it uses that file as to find the particular feature that will be executed.
+
+In addition to the list of features included, the file includes the following properties.
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| name | string | Name of the collection |
+| reference | string | Reference information of the repository and path where the code is stored. |
+| version | string | Version of the code. |
+
+In most cases the `feature-collection.json` file can be generated automatically at the moment of creating a release.
+
+## devcontainer.json properties
+
+Features are referenced in `devcontainer.json`  where the `features` tag consists of an array with the ordered list of features to be included in the container image.
+
+The properties are:
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| id | string | reference to the particular feature to be included. |
+| options | object | Type of the option |
+
+The `id` is the main reference point for how to find and download a particular feature. `id` can be defined in any of the following ways:
+
+| Type | Description |
+| :--- | :--- |
+| feature  | A simple name references a feature included with the application used to execute `devcontainer.json` |
+| organization/repository/{feature or collectionl/feature}/@version | Reference to a particular release in a repository |
+| https://<../URI/..>/devcontainer-features.tgz#{optional feature} | Direct reference to a file download. |
+| ./{local-path}  -or-  ../{local-path} | A path relative to devcontainer.json where a feature or feature collection can be found. |
+
+## Authoring
+
+Features can be authored in a number of languages, the most straight forward being bash scripts. If a feature is authored in a different language information about it should be included in the metadata so that users can make an informed choice about it.
+
+Reference information about the application required to execute the feature should be included in `feature.json` in the metadata section.
+
+Applications should default to `/bin/sh` for features that do not include this information.
+
+If the feature is included in a folder as part of the repository that contains `devcontainer.json` no other steps are necessary.
+
+## Release
+
+A release is created when the objective is to have other users use a feature.
+
+A release consists of the following:
+
+1.- Tar file with all the included files for the feature or feature collection.
+2.- A copy of the `feature.json` or `feature-collection.json` file that defines the contents of the tar file with additional information added to validate it:
+
+| Property | Type | Description |
+| :--- | :--- | :--- |
+| version | string | SemVer version of the release. |
+| checksum | string | Checksum of the tar file. |
```

Comment:
Adding the checksum of the tar file to the tar file itself is not possible. Are you considering having two files?

With NPM packages the version goes into the package and the checksum is computed on the package tar. We could add the checksum to the index file we have discussed and make the tar file the only release file.

Reply:
I think I wasn't clear, but the idea is that the file is included in the tar file so that during processing we have the information. We only add the checksum to the file that gets posted as a release next to it so that the download can be validated. I think it just simplifies things this way.